File: 128 Longest Consecutive Sequence/Python/Solution.py
from typing import List, Self


class Solution:
    def longestConsecutive(self: Self, nums: List[int]) -> int:
        l: int = len(nums)

        if l == 0:
            return 0
        
        nums.sort()
        maxLen: int = 1
        consecutiveLen: int = 1
        for i in range(1, l):
            if nums[i] == nums[i-1] + 1:
                consecutiveLen += 1
            elif nums[i] == nums[i-1]:
                continue
            else:
                if consecutiveLen > maxLen:
                    maxLen = consecutiveLen
                consecutiveLen = 1

        if consecutiveLen > maxLen:
            maxLen = consecutiveLen
        
        return maxLen

File: 128 Longest Consecutive Sequence/Java/Solution.java
import java.util.Arrays;

public class Solution {
    public int longestConsecutive(int[] nums) {
        int len = nums.length;

        if (len == 0) {
            return 0;
        }

        Arrays.sort(nums);
        int maxLen = 1;
        int consecutiveLen = 1;
        for (int i = 1; i < len; i++) {
            if (nums[i] == nums[i-1] + 1) {
                consecutiveLen++;
            }
            else if (nums[i] == nums[i-1]) {
                continue;
            }
            else {
                if (consecutiveLen > maxLen) {
                    maxLen = consecutiveLen;
                }
                consecutiveLen = 1;
            }
        }

        if (consecutiveLen > maxLen) {
            maxLen = consecutiveLen;
        }

        return maxLen;
    }
}
